source/contracts/extensions/MarketFeeCalculator.sol

```diff
@@ -3,45 +3,76 @@ pragma solidity ^0.4.13;
         if (_currentValidityBondInAttoeth != 0) {
             return _currentValidityBondInAttoeth;
         }
-        // TODO: get the real data for this
-        uint256 _indeterminateMarketsInPreviousWindow = 10;
-        // TODO: get the real data for this
-        uint256 _totalMarketsInPreviousWindow = 1000;
-        uint256 _previousTimestamp = _reportingWindow.getStartTime() - 1;
-        IUniverse _universe = _reportingWindow.getUniverse();
-        IReportingWindow _previousReportingWindow = _universe.getReportingWindowByTimestamp(_previousTimestamp);
+        IReportingWindow _previousReportingWindow = _reportingWindow.getPreviousReportingWindow();
+        uint256 _totalMarketsInPreviousWindow = _reportingWindow.getNumMarkets();
+        uint256 _indeterminateMarketsInPreviousWindow = _reportingWindow.getNumIndeterminateMarkets();
         uint256 _previousValidityBondInAttoeth = validityBondInAttoeth[_previousReportingWindow];
-        if (_previousValidityBondInAttoeth == 0) {
-            _previousValidityBondInAttoeth = 1 * 10 ** 16;
-        }
-        uint256 _targetIndeterminateMarketsPerHundred = 1;
-        _currentValidityBondInAttoeth = _previousValidityBondInAttoeth * _targetIndeterminateMarketsPerHundred * _totalMarketsInPreviousWindow / _indeterminateMarketsInPreviousWindow / 100;
+
+        _currentValidityBondInAttoeth = calculateValidityBond(_indeterminateMarketsInPreviousWindow, _totalMarketsInPreviousWindow, TARGET_INDETERMINATE_MARKETS_DIVISOR, _previousValidityBondInAttoeth);
         validityBondInAttoeth[_reportingWindow] = _currentValidityBondInAttoeth;
         return _currentValidityBondInAttoeth;
     }
 
-    function getTargetReporterGasCosts() constant public returns (uint256) {
-        // TODO: get number of registration tokens issued last period
-        // TODO: get target reporter count + wiggle room
-        // TODO: calculate estimated reporters per market
-        uint256 _estimatedReportsPerMarket = 10;
-        // TODO: figure out what the number actually is
-        uint256 _gasToReport = 100000;
-        // we double it to ensure we have more than enough rather than not enough
-        uint256 _estimatedReportingGas = _gasToReport * _estimatedReportsPerMarket * 2;
-        // TODO: multiply this by average gas costs of reporters historically
-        return _estimatedReportingGas;
+    function calculateValidityBond(uint256 _indeterminateMarkets, uint256 _totalMarkets, uint256 _targetIndeterminateMarketsDivisor, uint256 _previousValidityBondInAttoeth) constant public returns (uint256) {
+        if (_totalMarkets == 0) {
+            return DEFAULT_VALIDITY_BOND;
+        }
+        if (_previousValidityBondInAttoeth == 0) {
+            _previousValidityBondInAttoeth = DEFAULT_VALIDITY_BOND;
+        }
+        
+        uint256 _targetIndeterminateMarkets = _totalMarkets.div(_targetIndeterminateMarketsDivisor);
```

Comment:
If `_totalMarkets < 100` then this will evaluate to `0`.  Is that OK?  What if there are are 10 markets and 9 are indeterminate (90%), it feels like we shouldn't treat that like 0 indeterminate markets.

[Wolfram Alpha](http://www.wolframalpha.com/input/?i=a+%3C%3D+b%2Fc,+solve+for+b) to the rescue again (since I can' never remember inequality math):
```
if (_indeterminateMarkets.mul(_targetIndeterminateMarketDivisor) <= _targetIndeterminateMarkets) { ... }
```

Probably worth a comment indicating what is happening, since the multiplication of the divisor is a bit weird.

Reply:
`_totalMarkets` can be less than 100 without it necessarily evaluating to `0` since it's being divided by the target divisor. If the divisor is greater than the number of markets it is essentially the equivalent of saying we expect to get 0 indeterminate markets for that amount. (10 markets with a divisor of 20 means we expect 0.5 indeterminate markets, so 0 seems appropriate)

Reply:
Ah, so if we have a divisor of 100 (1% target) and 99 markets and 1 of those markets is indeterminate then we'll correctly hit the `else` branch because `(1 > 98/99)`.  So even though we rounded to 0, its fine because we'll still ultimately hit the correct code path.  👍